### ISSUE TITLE
Kotlin Native 0.8

### ISSUE THREAD

**Issue by RUSshy** [frol/completely-unscientific-benchmarks#61]
Hi,

Can u update kotlin native benchmark with latest version please ?

**Comment by frol** [comment]
Please, run the benchmark on your computer with the Kotlin version I use and Kotlin 0.8, and report what kind of performance boost you see.

**Comment by RUSshy** [comment]
all benchmark should be done using same config to make comparison, my PC is different than yours, and i can't run all benchmark

**Comment by frol** [comment]
My point is that I don't have a capacity to run every new release of every compiler in the benchmark to only confirm that it doesn't change anything. If you can compare the older Kotlin release performance with the new one and report that on your machine it speeds things up by 10%, I will rerun it on my machine and update the scoreboard. Otherwise, if the performance is the same, I can wait for the next release (ideally, 1.0).

**Comment by frol** [comment]
Well, I found some time today and re-run Kotlin/Native solution using 0.8 release... Ouch, they have managed to slow it down from 5.88s to 6.32s and the memory consumption has increased from 1.2MB to 7MB. I am not sure what I should do with these results now...

**Comment by RUSshy** [comment]
Thanks for taking time to do it

I also tried on windows with mingw

8.026s for 0.7
8.051s for 0.8.2

I was hoping they fix performance, looks like they don't care..

**Comment by RUSshy** [comment]
9.518s with 0.9

I don't know what they do, instead of fixing performance issue, they make it worse...

**Comment by barsan-ds** [comment]
@RUSshy did you have the time to test the new Beta release 1.3.x?

**Comment by barsan-ds** [comment]
My results are (on Linux):

v0.9
real    0m5,452s
user    0m5,452s
sys     0m0,000s

v1.3.10
real    0m5,549s
user    0m5,545s
sys     0m0,004s

No performance improvements yet.

**Comment by proyb6** [comment]
My results:
Source compile from master repo (5 Jan 2019)
Build Kotlin/Native infrastructure took about an hour
macOS Mojave on Macbook Pro Early 2015 (2.7 GHz Intel Core i5)

Has to update the code from:
```srand(time(null).toInt())```
to:
```srand(time(null).toUInt())```
to fix the error.

With Optimization flag (Worse?) took 26 seconds to compile:
real	0m8.931s
user	0m8.828s
sys	0m0.066s
Used 21MB+ memory

Without Optimization flag took 18 seconds to compile:
real	0m12.211s
user	0m12.116s
sys	0m0.047s
Used 10MB+ memory

You read it right, optimization flag version consumes more memory.

**Comment by frol** [comment]
@proyb6 hmm, but we don't compare compilation times here.

**Comment by proyb6** [comment]
Well, it’s not for the benchmark but for curious readers.

**Comment by frol** [comment]
@proyb6

> You read it right, optimization flag version consumes more memory.

There is nothing unusual that optimization process during compilation consumes more time and memory. This is the case for all compilers I know. The trade-off here is that the longer compilation time pays off with the faster run time of the resulting executable.

**Comment by proyb6** [comment]
I understand especially coming from Swift, Crystal, Rust and etc

**Comment by proyb6** [comment]
I see I was referring to the memory usage when the program is run which use 21MB (optimization build), not the compilation part. Thank for your time to reply.

**Comment by frol** [comment]
Ah, I see now! I think, I was misguided by "(Worse?)" comment.

Yeah, the overall trend to significant performance degradation is definitely not what I had expected with newer compiler releases.

**Comment by barsan-md** [comment]
For those still interested in Kotlin/Native, I've see [this](https://github.com/JetBrains/kotlin-native/commit/eb1221f465a5101a044c716f8751419fbecd01ea) commit that was really promising, so I tested the latest available release. It seems like there was a drastic performance improvement.
Best of 3 runs
v0.8
```
real    0m6,593s
user    0m6,586s
sys     0m0,008s
```
v1.3.50 - EAP
```
real    0m3,561s
user    0m3,553s
sys     0m0,008s
```

**Comment by RUSshy** [comment]
Nice, this looks promising for the future!

now they need:

- keep improving

- improve compile time

- add value type

Then we get perfect native language

**Comment by frol** [comment]
@barsan-md Thanks for the update! I will get the benchmark re-run in near future.

**Comment by mahdisml** [comment]
kotlin 1.4 :

Performance improvements

We’re continuously working to improve the overall performance of Kotlin/Native compilation and execution. In 1.4-M1, we offer you the new object allocator that works up to two times faster on some benchmarks. Currently, the new allocator is experimental and is not used by default; you can switch to it using the -Xallocator=mimalloc compiler option.

🤔🤔🤔

**Comment by frol** [comment]
FYI, I have re-run the benchmark on the same laptop I originally used with the latest released compilers (Kotlin Native 1.3.71). Most of the results got some subtle improvements. JVM (13 vs 8) got something like 10%-15% slower, so Kotlin JVM followed Java, but Kotlin Native (0.7 -> 1.3.71) improved from 5.88 seconds to 3.08 seconds! They are moving in the right direction, though it is going to be a long way :smile:
